cpp/src/arrow/compute/kernel.h

```diff
@@ -664,7 +664,7 @@ struct VectorKernel : public ArrayKernel {
 using ScalarAggregateConsume = std::function<void(KernelContext*, const ExecBatch&)>;
 
 using ScalarAggregateMerge =
-    std::function<void(KernelContext*, const KernelState&, KernelState*)>;
+    std::function<void(KernelContext*, KernelState&, KernelState*)>;
```

Comment:
If it can be moved, shouldn't you pass `KernelState&&` instead?

Reply:
Yes. Pass rvalue makes it more explicit. Will change.

Reply:
done